fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/ProgressiveLoanScheduleGenerator.java

```diff
@@ -103,11 +104,15 @@ public LoanScheduleModel generate(final MathContext mc, final LoanApplicationTer
 
         prepareDisbursementsOnLoanApplicationTerms(loanApplicationTerms);
 
+        final ArrayList<DisbursementData> disbursementDataList = new ArrayList<>(loanApplicationTerms.getDisbursementDatas());
+        disbursementDataList.sort(Comparator.comparing(DisbursementData::disbursementDate));
```

Comment:
Is disbursementDate a Date or a Timestamp? Is it possible that they get same "disbursementDate"s? If so, we should fall back to a secondary sort attribute like created date/etc.

Reply:
The `DisbursementData::disbursementDate` is a LocalDate. This method will handle the fallback situation.

Reply:
Can you pls explain how it would handle a secondary sorting in case there's a collision in date?

Reply:
I'm not sure what you mean by "secondary sorting" ? If this logic runs again?

Reply:
It could be easier to talk about the purpose of that code.
It's sorting the disbursement based on their effect date. Until there is no "actual date" it will use the expected date, and if disbursed it will use the actual date. The goal is just to apply disbursements in chronological order

Reply:
I mean what if there are two disbursements on the same date? Which one comes first, do we know?

Reply:
noted, do you think we could create a test-case to cover the "same disbursement day" behavior?

Reply:
yes, I added an extra case to the integration test and the EMI calculator to cover the same date scenario.